Improve Tree Traverser for use in CEFX

PhilMFischer opened the pull request:
- Updated the TreeTraverser.
- Improved comments.
- Added functionality to conditionally stop traversing children.
- Adjusted test cases to changes.
- Adjusted existing code to changes. others are dealt with default implementation on interface change.

Closes #414

Commit:
Updated Interface of Matcher - (Task #414)

- Added functionality to interface to decide on if child nodes should be
proecessed or not.
- Implement as default method always evaluating children if not
implemented in another way.
- Adjusted algorithm of the traverser
- Improved comments.

---
Task #414: Improve Tree Traverser Recursiveness

Commit:
Updated test case for tree traverser - (Task #414)

- Added test case to check the new functionality to decide if children
should be further evaluated.

---
Task #414: Improve Tree Traverser Recursiveness

Commit:
Merge remote-tracking branch 'origin/development' into feature/task_414-Improve_Tree_Traverse

andriikovalov-dlr commented on de.dlr.sc.virsat.model.edit/src/de/dlr/sc/virsat/model/concept/types/structural/tree/IBeanStructuralTreeTraverserMatcher.java:
```diff
@@ -31,5 +31,17 @@
+	 * @param isMatching the matching result which the traverser evaluated before.
+	 * @return true in case the children should be processed. false in case not.
+	 */
+	default boolean processChildren(IBeanStructuralElementInstance treeNode, boolean isMatching) {
```
Do we really want to pass `isMatching`? We have this information in the matcher, then pass it to the traverser, and then pass it back to the matcher. If this information is relevant, we can just check it with `isMatching` in the matcher. I would suggest we remove it to have a cleaner interface.

PhilMFischer replied:
I see the point with the interface. Still, in CEF is a case where we need to process the isMatching again in the processing method, which would result in a duplicate call and duplicate processing of the same method which is quite inefficient.

Since it is just an input parameter to the method i prefer leaving it in, since it does not need to be considered for the methods implementation if it has no value.

andriikovalov-dlr reviewed (changes requested):
I like this functionality, but I have a couple of remarks to the code

SaMuellerDLR commented on de.dlr.sc.virsat.model.edit/src/de/dlr/sc/virsat/model/concept/types/structural/tree/IBeanStructuralTreeTraverserMatcher.java:
```diff
@@ -31,5 +31,17 @@
+	 * @param isMatching the matching result which the traverser evaluated before.
+	 * @return true in case the children should be processed. false in case not.
+	 */
+	default boolean processChildren(IBeanStructuralElementInstance treeNode, boolean isMatching) {
```
Agree. It would be a bit long but I would propose `hasChildrenToProcess`

PhilMFischer replied:
Yes, changed it to continueTraverseChildren

Commit:
Changed interface due to review - (Task #414)

- renamed method in interface to be more precies.

---
Task #414: Improve Tree Traverser Recursiveness

SaMuellerDLR reviewed (approved):
Approved!
